src/pages/MyMove/PPMBooking/DateAndLocation.test.jsx

```diff
@@ -0,0 +1,340 @@
+            sitExpected: true,
+            expectedDepartureDate: '2022-07-04',
+          },
+        });
+
+        expect(updateMTOShipment).toHaveBeenCalledWith({ id: mockNewShipmentId });
+        expect(mockPush).toHaveBeenCalledWith(
+          generatePath(customerRoutes.SHIPMENT_PPM_ESTIMATED_WEIGHT_PATH, {
+            moveId: mockMoveId,
+            mtoShipmentId: mockNewShipmentId,
+          }),
+        );
+      });
+    });
+  });
+
+  describe('editing an existing PPM shipment', () => {
+    it('renders the heading and form with shipment values', async () => {
+      render(<DateAndLocation {...fullShipmentProps} />);
+
+      expect(screen.getByRole('heading', { level: 1 })).toHaveTextContent('PPM date & location');
+
+      const postalCodes = screen.getAllByLabelText('ZIP');
+      const secondaryPostalCodes = screen.getAllByLabelText('Second ZIP');
+
+      await waitFor(() => {
+        expect(screen.getByLabelText('When do you plan to start moving your PPM?')).toHaveValue('31 Dec 2022');
+      });
+
+      expect(postalCodes[0]).toHaveValue('20002');
+      expect(postalCodes[1]).toHaveValue('20004');
+      expect(secondaryPostalCodes[0]).toHaveValue('20003');
+      expect(secondaryPostalCodes[1]).toHaveValue('20005');
+      expect(screen.getAllByLabelText('Yes')[2]).toBeChecked();
+    });
+
+    it('routes back to the home page screen when back is clicked', async () => {
+      const selectShipmentType = generatePath(generalRoutes.HOME_PATH);
+
+      render(<DateAndLocation {...defaultProps} {...fullShipmentProps} />);
+
+      userEvent.click(screen.getByRole('button', { name: 'Back' }));
+
+      expect(mockPush).toHaveBeenCalledWith(selectShipmentType);
+    });
+
+    it('displays an error alert when the update shipment fails', async () => {
+      patchMTOShipment.mockRejectedValueOnce('fatal error');
+
+      render(<DateAndLocation {...fullShipmentProps} />);
+
+      userEvent.click(screen.getByRole('button', { name: 'Save & Continue' }));
+
+      await waitFor(() => {
+        expect(patchMTOShipment).toHaveBeenCalledWith(
+          fullShipmentProps.mtoShipment.id,
+          {
+            id: fullShipmentProps.mtoShipment.id,
+            moveTaskOrderID: mockMoveId,
+            shipmentType: 'PPM',
+            ppmShipment: {
+              id: fullShipmentProps.mtoShipment.ppmShipment.id,
+              pickupPostalCode: '20002',
+              destinationPostalCode: '20004',
+              hasSecondaryPickupPostalCode: true,
+              secondaryPickupPostalCode: '20003',
+              hasSecondaryDestinationPostalCode: true,
+              secondaryDestinationPostalCode: '20005',
+              sitExpected: true,
+              expectedDepartureDate: '2022-12-31',
+            },
+          },
+          fullShipmentProps.mtoShipment.eTag,
+        );
+
+        expect(screen.getByText('There was an error attempting to updateq your shipment.')).toBeInTheDocument();
```

Comment:
Small typo causing this test to fail.
```suggestion
        expect(screen.getByText('There was an error attempting to update your shipment.')).toBeInTheDocument();
```